src/StreamJsonRpc/CommonMethodNameTransforms.cs

```diff
@@ -45,7 +47,10 @@ public static Func<string, string> Prepend(string prefix)
         {
             return name => name;
         }
-
-        return name => prefix + name;
+        else
+        {
+            string localPrefix = prefix;
```

Comment:
This had a closure being allocated even on the fast path when prefix.Length == 0.